### ISSUE TITLE
117 unlinked exchanges in ecoinvent 3.7.1 consequential 

### ISSUE THREAD

**Issue by mfastudillo** [brightway-lca/brightway2-io#96]
Hi

using bw2io version (0, 7, 12, 1) if I try to import ecoinvent consequential, I get 117 unlinked exchanges 

    ei371consequential=bw.SingleOutputEcospold2Importer(ecoinvent_371clca_path,"ei_371con")
    ei371consequential.apply_strategies()
    ei371consequential.statistics()

I need to run `.drop_unlinked(i_am_reckless=True)` (XD) to save it 
    
    ei371consequential.drop_unlinked(i_am_reckless=True)

something to keep in mind

**Comment by GuillaumeBourgaultEcoinvent** [comment]
This is a known issue.  We are changing the way the datasets are calculated.  Unfortunately, this change will not come on time for 3.8, but this issue will not be present from 3.9 onwards.  
These exchanges can be safely ignored.

**Comment by cmutel** [comment]
@mfastudillo Is this still true on 3.8? Do these exchanges have non-zero amounts? We have cleaned up releases in the past, we can do this again if there is a pattern we can filter against.

**Comment by cedric-roussel** [comment]
I didn't have this issue when importing 3.8 consequential.

To apply filtering on 3.7.1 use compartment == "social"

117 unlinked exchanges
  Type biosphere: 3 unique unlinked exchanges

https://ecoinvent.org/the-ecoinvent-database/data-releases/ecoinvent-3-7-1/#!/known-issues
Consequential system model issues
Three elementary exchanges are found in the compartment “social”. These exchanges can be ignored, both at the unit process and the inventory level, as ecoinvent does not yet account for social impacts.

**Comment by mfastudillo** [comment]
> 
> 
> @mfastudillo Is this still true on 3.8? Do these exchanges have non-zero amounts? We have cleaned up releases in the past, we can do this again if there is a pattern we can filter against.

I imported 3.8 consequential yesterday with bw2data version (4, 0, 'DEV4') and bw2io version (0, 9, 'DEV3') and I did not have this problem.

**Comment by cmutel** [comment]
@cedric-roussel Please see if ed00d35a3303a659ddb8e2d070060af72f1307ad works for you.

**Comment by cedric-roussel** [comment]
Logic is correct, but it seems you won't have compartments in strategies, if that's where you want to apply this code.
The "categories" array might need to be provisioned somewhere here:
https://github.com/ecoinvent/brightway2-io/blob/master/bw2io/extractors/ecospold2.py#L413.

Alternatively, the filtering can be applied earlier, untested, maybe something like
 ```diff --git a/bw2io/extractors/ecospold2.py b/bw2io/extractors/ecospold2.py
index a578a4e..53f33d6 100644
--- a/bw2io/extractors/ecospold2.py
+++ b/bw2io/extractors/ecospold2.py
@@ -191,6 +191,7 @@ class Ecospold2DataExtractor(object):
                 cls.extract_exchange(exc)
                 for exc in stem.flowData.iterchildren()
                 if "parameter" not in exc.tag
+                and not (hasattr(exc,"compartment") and exc.compartment.compartment.text == "social")
             ],
             "filename": os.path.basename(filename),
             "location": stem.activityDescription.geography.shortname.text,

```

Since this is solved in 3.8 and I don't know how many brightway users are actually using the consequential 3.7.1 release... Eventually it's OK to just "be reckless" in that case ? :)

Thanks @cmutel

**Comment by cmutel** [comment]
People are still using 3.7 consequential. Fixed in 9c9d4e01a2f20edb9badcb3be364d5cf9c7b47d1.
